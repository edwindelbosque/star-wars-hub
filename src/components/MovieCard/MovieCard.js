import React from 'react';
import './MovieCard.scss';
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types';
import one from '../../images/1.jpg';
import two from '../../images/2.jpg';
import three from '../../images/3.jpg';
import four from '../../images/4.jpg';
import five from '../../images/5.jpg';
import six from '../../images/6.jpg';
import seven from '../../images/7.jpg';

const images = {
  1: [one, 'https://www.youtube.com/watch?v=bD7bpG-zDJQ', 'https://www.amazon.com/Star-Wars-Episode-Phantom-Widescreen/dp/B00003CX5P'],
  2: [two, 'https://www.youtube.com/watch?v=gYbW1F_c9eM', 'https://www.amazon.com/Star-Wars-Episode-Attack-Widescreen/dp/B00006HBUJ'],
  3: [three, 'https://www.youtube.com/watch?v=5UnjrG_N8hU', 'https://www.amazon.com/Star-Wars-Episode-Revenge-Widescreen/dp/B00005JLXH'],
  4: [four, 'https://www.youtube.com/watch?v=MpkrMqmmy5k', 'https://www.amazon.com/Star-Wars-Episode-IV-Limited/dp/B000FQJAIW'],
  5: [five, 'https://www.youtube.com/watch?v=xESiohGGP7g', 'https://www.amazon.com/Star-Wars-Empire-Strikes-Limited/dp/B000FQJAJG'],
  6: [six, 'https://www.youtube.com/watch?v=MYD_xxY5wEI', 'https://www.amazon.com/Star-Wars-Episode-VI-Limited/dp/B000FQVX78'],
  7: [seven, 'https://www.youtube.com/watch?v=sGbxmsDFVnE', 'https://www.amazon.com/Star-Wars-Episode-2Blu-Ray-subtitles/dp/B01EFU2GKC']
}

const MovieCard = ({ title, episode, releaseYear, filmId, getDetails }) => {
  return (
    <section
      className={`movie-card movie-card-${episode}`}
      classNameid={filmId}
      onClick={(e) => getDetails(filmId)}
    >
      <img
        src={images[episode][0]}
        className='poster-image'
        alt={`Episode ${episode} movie poster`}
      />
      <h3 onClick={(e) => console.log(e.target.id)}>{title}</h3>
      <p className='movie-card-text'>Episode {episode}</p>
      <p className='movie-card-text'>{releaseYear.slice(0, 4)}</p>
      <a className='text-link' href={images[episode][2]}>Buy Movie</a>
      <a className='text-link' href={images[episode][1]}>Watch Trailer</a>
      <Link to={`/movies/${episode}`} className='character-link'>View Characters</Link>
    </section>
  )
}

MovieCard.propTypes = {
  title: PropTypes.string,
  episode: PropTypes.number,
  releaseYear: PropTypes.string,
  filmId: PropTypes.number,
  getDetails: PropTypes.func
};

export default MovieCard;