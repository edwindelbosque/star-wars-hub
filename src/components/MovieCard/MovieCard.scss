.movie-card {
    border-radius: 2px 2px 6px 6px;
    background: linear-gradient(to top left, rgba(114, 114, 114, 0.158), #2d2d2d00);
    display: flex;
    border: solid rgb(143, 143, 143) 1px;
    box-shadow: 0 0 3px rgb(130, 130, 130);
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-between;
    transition: .2s;
}

.poster-image {
    width: 100%;
    border-bottom: 1px solid rgb(72, 72, 72);
}

h3 {
    font-size: 28px;
    margin: 20px 30px 10px 30px;
    border-bottom: 1px solid rgba(155, 155, 155, 0.233);
    width: 85%;
    padding-bottom: 5px;
    text-align: left;
}

.movie-card-text {
    margin: 5px 30px;
    color: rgb(205, 205, 205);
    font-weight: 700;
    font-size: 16px;
}

.character-link {
    border-top: 1px solid rgb(218, 218, 218);
    width: 85%;
    height: 50px;
    margin: 5px 30px;
    font-size: 23px;
    padding-top: 12px;
    text-decoration: none;
    color: rgb(189, 189, 189);
    transition: .2s;

    &:hover {
        border-top: 1px solid #ff0000;
        transition: .2s;
        color: rgb(255, 96, 96);
    }

    &:active {
        transition: .2s;
        color: rgb(185, 78, 78);
        border-top: 1px solid #a54343;
    }
}

.movie-card-1,
.movie-card-2,
.movie-card-3 {
  h3 {
    color: rgb(255, 126, 126);
  }

  &:hover {
    box-shadow: 0 0 15px rgb(255, 95, 95);
    transition: .2s;
  }
  &:active {
    box-shadow: 0 0 30px rgb(255, 108, 108);
    transition: .2s;
  }
}

.movie-card-4,
.movie-card-5,
.movie-card-6 {
  h3 {
    color: rgb(126, 225, 255);
  }

  &:hover {
    box-shadow: 0 0 15px rgb(95, 204, 255);
    transition: .2s;
  }
  &:active {
    box-shadow: 0 0 30px rgb(125, 196, 255);
    transition: .2s;
  }
}

.movie-card-7 {
  h3 {
    color: rgb(255, 165, 126);
  }
  
  &:hover {
    box-shadow: 0 0 15px rgb(255, 177, 141);
    transition: .2s;
  }
  &:active {
    box-shadow: 0 0 30px rgb(255, 201, 144);
    transition: .2s;
  }
}