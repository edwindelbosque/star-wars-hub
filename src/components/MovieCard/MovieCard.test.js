import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from './MovieCard';

describe('MovieCard', () => {
  const getDetails = jest.fn();
  it('component should match the snapshot', () => {
    const wrapper = shallow(
      <MovieCard
        key={3}
        title={'The Phantom Menace'}
        episode={3}
        releaseYear={'2099'}
        filmId={3}
        getDetails={getDetails}
      />
    )
    expect(wrapper).toMatchSnapshot();
  });
})