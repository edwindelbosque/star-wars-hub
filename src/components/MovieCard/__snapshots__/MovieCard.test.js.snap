// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard component should match the snapshot 1`] = `
<section
  className="movie-card movie-card-3 {filmId}"
  onClick={[Function]}
>
  <img
    alt="Episode 3 movie poster"
    className="poster-image"
    src="3.jpg"
  />
  <h3
    onClick={[Function]}
  >
    The Phantom Menace
  </h3>
  <p
    className="movie-card-text"
  >
    Episode 
    3
  </p>
  <p
    className="movie-card-text"
  >
    2099
  </p>
  <a
    className="text-link"
    href="https://www.amazon.com/Star-Wars-Episode-Revenge-Widescreen/dp/B00005JLXH"
    rel="noopener noreferrer"
    target="_blank"
  >
    Buy Movie
  </a>
  <a
    className="text-link"
    href="https://www.youtube.com/watch?v=5UnjrG_N8hU"
    rel="noopener noreferrer"
    target="_blank"
  >
    Watch Trailer
  </a>
  <Link
    className="character-link"
    to="/movies/3"
  >
    View Characters
  </Link>
</section>
`;
